engine/component.ftl

```diff
@@ -283,44 +283,45 @@
     [#return profile[key]!{} ]
 [/#function]
 
-[#function invokeComponentMacro occurrence resourceGroup type qualifiers=[] parent={} includeShared=false ]
+[#function invokeComponentMacro occurrence resourceGroup entrance="" qualifiers=[] parent={} includeShared=true ]
     [#local placement = (occurrence.State.ResourceGroups[resourceGroup].Placement)!{} ]
     [#if placement?has_content]
         [#local macroOptions = [] ]
-        [#list asArray(qualifiers) as qualifier]
+        [#list qualifiers as qualifier]
             [#local macroOptions +=
                 [
-                    [placement.Provider, occurrence.Core.Type, resourceGroup, placement.DeploymentFramework, type, qualifier],
-                    [placement.Provider, occurrence.Core.Type, placement.DeploymentFramework, type, qualifier],
-                    [placement.Provider, resourceGroup, placement.DeploymentFramework, type, qualifier]
+                    [placement.Provider, occurrence.Core.Type, resourceGroup, placement.DeploymentFramework, entrance ] + asArray(qualifier),
+                    [placement.Provider, occurrence.Core.Type, placement.DeploymentFramework, entrance ] + asArray(qualifier),
+                    [placement.Provider, resourceGroup, placement.DeploymentFramework, entrance ] + asArray(qualifier),
+                    [placement.Provider, resourceGroup, placement.DeploymentFramework  ] + asArray(qualifier)
                 ]]
         [/#list]
 
         [#local macroOptions +=
             [
-                [placement.Provider, occurrence.Core.Type, resourceGroup, placement.DeploymentFramework, type],
-                [placement.Provider, occurrence.Core.Type, placement.DeploymentFramework, type],
-                [placement.Provider, resourceGroup, placement.DeploymentFramework, type]
+                [placement.Provider, occurrence.Core.Type, resourceGroup, placement.DeploymentFramework, entrance],
+                [placement.Provider, occurrence.Core.Type, placement.DeploymentFramework, entrance],
+                [placement.Provider, resourceGroup, placement.DeploymentFramework, entrance],
+                [placement.Provider, resourceGroup, placement.DeploymentFramework ]
             ]]
-
         [#if includeShared ]
-            [#list asArray(qualifiers) as qualifier]
+            [#list qualifiers as qualifier]
```

Comment:
The main reason to move it was to allow for an array to be passed as a qualifier, this would allow a qualifier to add two levels of qualification instead of just the single level. 

I've updated it though to do the asArray for the first part and then asFlattenedArray remove empty to remove the empty string